src/js/player.js

```diff
@@ -314,6 +314,11 @@ vjs.Player.prototype.loadTech = function(techName, source){
 
   var techReady = function(){
     this.player_.triggerReady();
+
+    // add vjs-autoplay to hide poster and big-play-button if autoplay:true
+    if (this.options_['autoplay']) {
+      this.addClass('vjs-autoplay');
```

Comment:
Two questions:
- Why not just use an attribute selector to match when the `autoplay` attribute is actually present on the player div?
- If we do go this route, don't we need some equivalent code that removes the autoplay class when the attribute is removed?